test/test_dependencies.py

```diff
@@ -189,3 +189,43 @@ def test_colcon_defaults(tmpdir):
     assert 'ros-dashing-ament-cmake' not in result
     assert 'ros-dashing-rclcpp' not in result
     assert 'ros-dashing-rclpy' in result
+
+
+@uses_docker
+def test_dummy_skip_rosdep_keys_doesnt_exist_pkg(tmpdir):
+    ws = Path(str(tmpdir))
+    pkg_xml = Path(ws) / 'src' / 'dummy' / 'package.xml'
+    pkg_xml.parent.mkdir(parents=True)
+    pkg_xml.write_text(CUSTOM_KEY_PKG_XML)
+    client = DockerClient()
+    platform = Platform(arch='aarch64', os_name='ubuntu', ros_distro='dashing')
+    out_script = ws / rosdep_install_script(platform)
+
+    skip_keys = ['definitely_does_not_exist']
+    gather_rosdeps(client, platform, workspace=ws, skip_rosdep_keys=skip_keys)
+    result = out_script.read_text().splitlines()
+    expected = [
```

Comment:
We should not implement those type of tests, because they'll break as soon as the install script is modified.

This anti-pattern is called "change detector tests", see https://testing.googleblog.com/2015/01/testing-on-toilet-change-detector-tests.html for details.

Instead, let's make sure that the script does not contain "definitely_does_not_exist". This does the same thing, but will not fail when the install script structure will be modified!

Reply:
Actually, the test will throw an exception normally. Checking the contents is unecessary, the test is that the call does not throw.